src/java/org/apache/cassandra/config/Config.java

```diff
@@ -371,15 +427,21 @@
 
     public final ReplicaFilteringProtectionOptions replica_filtering_protection = new ReplicaFilteringProtectionOptions();
 
-    public volatile Long index_summary_capacity_in_mb;
-    public volatile int index_summary_resize_interval_in_minutes = 60;
+    @Replaces(oldName = "index_summary_capacity_in_mb", converter = Converter.MegabytesDataStorageConverter.class, deprecated = true)
+    public DataStorage index_summary_capacity;
+    @Replaces(oldName = "index_summary_resize_interval_in_minutes", converter = Converter.MinutesDurationConverter.class, deprecated = true)
+    public volatile CassandraDuration index_summary_resize_interval = new CassandraDuration("60m");
 
-    public int gc_log_threshold_in_ms = 200;
-    public int gc_warn_threshold_in_ms = 1000;
+    @Replaces(oldName = "gc_log_threshold_in_ms", converter = Converter.MillisDurationConverter.class, deprecated = true)
+    public CassandraDuration gc_log_threshold = new CassandraDuration("200ms");
+    @Replaces(oldName = "gc_warn_threshold_in_ms", converter = Converter.MillisDurationConverter.class, deprecated = true)
+    public CassandraDuration gc_warn_threshold = new CassandraDuration("1s");
 
     // TTL for different types of trace events.
-    public int tracetype_query_ttl = (int) TimeUnit.DAYS.toSeconds(1);
-    public int tracetype_repair_ttl = (int) TimeUnit.DAYS.toSeconds(7);
+    @Replaces(oldName = "tracetype_query_ttl", converter = Converter.SecondsDurationConverter.class, deprecated=true)
+    public CassandraDuration trace_type_query_ttl = new CassandraDuration("86400s");
```

Comment:
```suggestion
    public CassandraDuration trace_type_query_ttl = new CassandraDuration("1d");
```

Reply:
Those were intentionally kept with the old value but when I update the VTs and the JMX to use converter I will improve them according to your suggestions. Thanks